src/main/java/net/mcreator/ui/validation/validators/PointOfInterestValidator.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2022, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.validation.validators;
+
+import net.mcreator.element.parts.MItemBlock;
+import net.mcreator.minecraft.ElementUtil;
+import net.mcreator.ui.MCreator;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.ui.minecraft.MCItemHolder;
+import net.mcreator.ui.validation.Validator;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class PointOfInterestValidator implements Validator {
+
+	private final MCreator mcreator;
+	private final MCItemHolder holder;
+	private final MCItemHolder exception;
+	private final List<MItemBlock> vanillaPointOfInterest;
+
+	public PointOfInterestValidator(MCreator mcreator, MCItemHolder holder, MCItemHolder exception) {
```

Comment:
This class can be replaced with
```java
new UniqueNameValidator("Point of interest",
			e -> e.getBlock().getUnmappedValue(),
			() -> ElementUtil.loadAllPointOfInterest(mcreator.getWorkspace()),
			vanillaPointOfInterest,
			null)
```
Here `vanillaPointOfInterest` contains all blocks/items listed later in this file (their values should not be mapped registry names but unmapped data list entry names, the validation did not take them into account due to this bug).

Reply:
So you're saying I should delete the `PointOfInterestValidator` class? Also, could you explain what you mean by the fact that the blocks in `vanillaPointOfInterest` should not be mapped with registry names, but unmapped data list entry names?

Reply:
At this moment, you're specifying forbidden blocks like this:
`new MItemBlock(mcreator.getWorkspace(), "bed")`
However, `blocksitems.yaml` data list does not contain any entry with name `bed`. To make this work as you intended, you need to replace registry names with entry names:
`new MItemBlock(mcreator.getWorkspace(), "Blocks.BED")`
For the bed block particularily, you'd need to add all 16 bed types to the forbidden blocks list.
>So you're saying I should delete the `PointOfInterestValidator` class?

Now that `UniqueNameValidator` can accept a unique identifier string from any type of components, ~this is excess code~ you just don't have to make another class for your very specific use case.

Reply:
I tried the code you gave me, but it doesn't work. the `e` doesn't have a `getBlock` method, and it has other issues as well.

Reply:
Ah, sorry about that. Could you try again now?
>it has other issues as well

>Like what?

Reply:
It says that `vanillaPointOfInterest` but be a List<String> and right now, it is a List<MItemBlock>